p2p/server.go

```diff
@@ -132,15 +132,24 @@ func (s *peer) InitProtocolForDID(DID *identity.DID) {
 }
 
 func (s *peer) runDHT(ctx context.Context, bootstrapPeers []string) error {
-	s.dht = dht.NewDHT(ctx, s.host, ds.NewMapDatastore())
 	log.Infof("Bootstrapping %s\n", bootstrapPeers)
 
 	for _, addr := range bootstrapPeers {
-		iaddr, _ := ipfsaddr.ParseString(addr)
-		pinfo, _ := pstore.InfoFromP2pAddr(iaddr.Multiaddr())
-		if err := s.host.Connect(ctx, *pinfo); err != nil {
+		multiaddr, _ := ma.NewMultiaddr(addr)
+		p, err := libp2pPeer.AddrInfoFromP2pAddr(multiaddr)
+		if err != nil {
+			log.Info(err)
+			continue
+		}
+
+		s.host.Peerstore().AddAddrs(p.ID, p.Addrs, peerstore.PermanentAddrTTL)
+		err = s.host.Connect(ctx, *p)
```

Comment:
I think we could make all `Connect` calls in pararell with go-routines 
Similar to : https://github.com/ipfs/go-ipfs/blob/master/docs/examples/go-ipfs-as-a-library/main.go#L132

For improving performance.

Currently it doesn't matter that much, because every node uses our pre-defined bootstrap nodes.

Reply:
In either case it waits until all the go routines have filled the waitgroup. Internally already the routing table is refreshed every 5 mins using their own go routine.
I rather take the initial hit so we know that when the node is up, the routing table is filled with as much as relevant peers as possible.

Reply:
Yes, but i think establishing the connection can take some time, especially if bootstrap peers are not reachable and it is a timeout. 

I think if you do it in parallel with a waitinggroup the max time is equal to the peer with the longest connection establishment/timout instead of the sum of all connection establishments.

But in our case it doesn't matter yet, because we don't provide 10 bootstrap peers with 4 peers maybe not reachable in it.